### PR TITLE
updated serialization folder

### PR DESCRIPTION
**Pull request by haripriya9647**
## Description
added inline docs for methods and class
## Relevant Issue
Issue #269  the issue here

## Affected Dependencies
None

## How has this been tested?
- Describe the tests that you ran to verify your changes.
- Provide instructions so we can reproduce.
- List any relevant details for your test configuration.

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/NimbleEdge/EnvisEdge/blob/main/CONTRIBUTING.md) and [Code of Conduct](https://github.com/NimbleEdge/EnvisEdge/blob/main/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [NimbleEdge Styleguide](https://numpydoc.readthedocs.io/en/latest/format.html)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/NimbleEdge/EnvisEdge/labels)
- [x] My changes are covered by tests

### CODE REVIEW

File: fedrec/serialization/serializable_interface.py [hunk 1]
```diff
@@ -32,12 +32,18 @@ def __init__(self) -> None:
 
     @abstractmethod
     def serialize(self):
+        """
+        Serializes the data object into another format
+        """
         raise NotImplementedError()
```
Comment (divyanshugit) [inline fedrec/serialization/serializable_interface.py:38]: If it's not implemented then you don't have describe it

**Review by divyanshugit — CHANGES_REQUESTED** [review]

**Comment by divyanshugit** [comment]
For `serialization_strategy.py`, please explain the JSON serialization strategy.

File: fedrec/serialization/serialization_strategy.py [hunk 2]
```diff
@@ -8,10 +8,16 @@ class SerializationStrategy(registry.Registrable, ABC):
 
     @abstractmethod
     def parse(self, obj):
+        """
+        Serializes a python object to json.
+        """
```
Comment (divyanshugit) [inline fedrec/serialization/serialization_strategy.py:13]: Remove the comments, if it is not implemented.

File: fedrec/serialization/serialization_strategy.py [hunk 3]
```diff
@@ -8,10 +8,16 @@ class SerializationStrategy(registry.Registrable, ABC):
...
         raise NotImplementedError()
 
     @abstractmethod
     def unparse(self, obj):
+        """
+        Deserializes the json object to python object.
+        """
```
Comment (divyanshugit) [inline fedrec/serialization/serialization_strategy.py:20]: Same as above

**Review by divyanshugit — CHANGES_REQUESTED** [review]

**Comment by divyanshugit** [comment]
It will be great if you can add an example use case of method `append_type` in `serializable_interface.py `.  and You can explain a bit more about the Abstract class for serializers and deserializers.

**Review by ramesht007 — CHANGES_REQUESTED** [review]
So, the basic idea of `Serializable` class that it is the base class of all the data formats that need to be serialised that said the `serialise or deserialise attribute functions` check for this class as a base class, that is this class is serialisable or not.  
If they are child class of  `Serializable` then further process is continued.

Can you write this also in the coments.
thanks

**Comment by haripriya9647** [comment]
@ramesht007 pushed my commit as per your suggestions. please have a look

File: fedrec/serialization/serializable_interface.py [hunk 4]
```diff
@@ -13,12 +13,17 @@ def is_primitives(obj):
 
 class Serializable(Registrable, ABC):
     """
-    Abstract class for serializers and deserializers.
+    Serializable is the abstract base class for all the data
+    formats that needs to be serialized and deserialized.
     In simple terms, the serializable is converting a
```
Comment (ramesht007) [inline fedrec/serialization/serializable_interface.py:18]: Can you add this?
`serializable` is a parent class that ensures that the child classes can be serialized.
thanks

**Review by ramesht007 — CHANGES_REQUESTED** [review]

**Review by ramesht007 — APPROVED** [review]
LGTM!

**Pull request merged**